### PR TITLE
feat: enable csrf protection

### PR DESCRIPTION
**Pull request by mabreuortega**
## Description

Protect update operations with CSRF Tokens

## Testing

Describe how you tested your changes and/or give instructions to the reviewers on how they can verify them.

1- Go to https://d1qtnn3brrj5l2.cloudfront.net/admin
2- Perform any update operation
3- Service will go and fetch the CSRF Token 
4- Attach the token with the update request

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: backend/src/lib/api/middleware/csrf-handler.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { NextFunction, Request, Response } from "express";
+import { doubleCsrf } from "csrf-csrf";
+import cookieParser from "cookie-parser";
+
+const csrfSecret = process.env.CSRF_SECRET ?? Math.random().toString();
+const cookiesSecret = process.env.COOKIES_SECRET ?? Math.random().toString();
```
[the remark this answers is not in the document]
Comment (mabreuortega) [reply backend/src/lib/api/middleware/csrf-handler.ts:11]: this is not sensitive and it is fine to use a weak generator

File: frontend/src/services/__tests__/BackendService.test.ts [hunk 2]
```diff
@@ -13,6 +13,21 @@
...
 let windowSpy: any;
 
+function mockCsrfToken() {
+    API.get = jest.fn().mockReturnValueOnce({ token: "csrf-token-value" });
+}
+
+function expectedAuthenticatedRequest(obj: any = {}) {
+    return expect.objectContaining({
+        headers: {
+            Authorization: "Bearer eyJhbGciOiJIUzI1NiIsInR5c",
```
[the remark this answers is not in the document]
Comment (mabreuortega) [reply frontend/src/services/__tests__/BackendService.test.ts:23]: This is test data

File: backend/src/lib/api/middleware/csrf-handler.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        httpOnly: true,
+        secure: true,
+        signed: true,
+        path: "/",
+    },
+    size: 64,
+    ignoredMethods: ["GET", "HEAD", "OPTIONS"],
+});
+
+const csrfCookieParser = cookieParser(cookiesSecret);
```
[the remark this answers is not in the document]
Comment (mabreuortega) [reply backend/src/lib/api/middleware/csrf-handler.ts:28]: by design

**Review by denisquesada — APPROVED** [review]
LGTM. Good Work!

**Pull request merged**